tests/sles4sap/sap_deployment_automation_framework/prepare_ssh_config.pm

```diff
@@ -17,44 +17,40 @@ use sles4sap::azure_cli qw(az_keyvault_list);
 use sles4sap::sap_deployment_automation_framework::inventory_tools;
 use sles4sap::sap_deployment_automation_framework::deployment qw(sdaf_ssh_key_from_keyvault);
 use sles4sap::sap_deployment_automation_framework::naming_conventions
-  qw( get_sdaf_inventory_path
-  convert_region_to_short
-  get_workload_vnet_code
-  $sut_private_key_path
+  qw( get_sdaf_inventory_path get_sdaf_config_path convert_region_to_short get_workload_vnet_code $sut_private_key_path
   generate_resource_group_name);
 
 sub run {
     my ($self, $run_args) = @_;
     select_serial_terminal;
-
-    # Connect serial to Deployer VM to get inventory file
-    connect_target_to_serial();
+    my $env_code = get_required_var('SDAF_ENV_CODE');
+    my $sdaf_region_code = convert_region_to_short(get_required_var('PUBLIC_CLOUD_REGION'));
+    my $sap_sid = get_required_var('SAP_SID');
+    my $workload_vnet_code = get_workload_vnet_code(), sap_sid => $sap_sid;
 
     my $jump_host_user = get_required_var('REDIRECT_DESTINATION_USER');
     my $jump_host_ip = get_required_var('REDIRECT_DESTINATION_IP');
+    my $inventory_path = get_sdaf_inventory_path(env_code => $env_code, sdaf_region_code => $sdaf_region_code,
+        vnet_code => $workload_vnet_code, sap_sid => $sap_sid);
+    my $private_key_src_path = get_sdaf_config_path(deployment_type => 'sap_system', env_code => $env_code,
+        sdaf_region_code => $sdaf_region_code, sap_sid => $sap_sid, vnet_code => $workload_vnet_code) . '/sshkey';
+
+    # Connect serial to Deployer VM to get inventory file
+    connect_target_to_serial();
 
-    my $inventory_path = get_sdaf_inventory_path(
-        env_code => get_required_var('SDAF_ENV_CODE'),
-        sdaf_region_code => convert_region_to_short(get_required_var('PUBLIC_CLOUD_REGION')),
-        vnet_code => get_workload_vnet_code(),
-        sap_sid => get_required_var('SAP_SID')
-    );
     my $inventory_data = read_inventory_file($inventory_path);
     # From now on all commands will be executed on worker VM
     disconnect_target_from_serial();
 
+    # Download ssh private key for accessing SUTs
+    my $scp_cmd = join(' ', 'scp ', $ssh_opt, "$jump_host_user\@$jump_host_ip:$private_key_src_path", $sut_private_key_path);
+    assert_script_run($scp_cmd);
+
     # Share inventory data between all tests
     $run_args->{sdaf_inventory} = $inventory_data;
     # Create console redirection data
     $run_args->{redirection_data} = create_redirection_data(inventory_data => $inventory_data);
 
-    my @workload_key_vault = @{az_keyvault_list(
```

Comment:
This won't work anymore as keyvaults and storage access is blocked from outside world. 
https://github.com/Azure/sap-automation/issues/704